spec/30-processing-model.md

```diff
@@ -0,0 +1,64 @@
+   dash (`-`)), the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`. 
+    2. If the version number is higher than supported by the tracing implemenation, the implementation uses the format defined in this specification to parse `trace-id` and `parent-id`. The tracing system will only parse `flags` values supported by the current version of  this specification and ignore all other values. If parsing fails, the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`. 
+3. When the version number is supported by the tracing system it validates `trace-id` and `parent-id`.
+    1. If either `trace-id`, `parent-id` or `flags`  are invalid,  the tracing system creates a new `traceparent` header and removes all entries from `tracecontext`.
+4. After the system has validate the traceparent it validates the tracestate
+*NOTE* : How do we do this exactly?
+
+5. Tracing system modifies the `traceparent` header. 
+    1. **Update `parent-id`**. The value of property `parent-id` can be set to the
+   new value representing the ID of the current operation. This is the most
+   typical mutation and should be considered a default.
+*NOTE* : This definition is not entirely clear.   
+    2. **Indicate recorded state**. The value of `recorded` flag of `trace-flags`
+    may be set to `1` if it had value `0` before or vice versa. `parent-id` MUST
+    be set to the new value with the `recorded` flag update. See details of
+    `recorded` flag for more information on how this flag is recommended to be
+    used.
+    3. **Update `recorded`**. The value of `recorded` reflects the caller's
+    recording behavior: either the trace data were dropped or may have been
+    recorded out-of-band. This mutation gives the downstream tracer information
+    about the likelihood its parent's information was recorded.
+6. Tracing System modifies the `tracestate` header
+    1. **Update key value**. The value of any key can be updated. Modified keys MUST
+     be moved to the beginning of the list. This is the most common mutation
+     resuming the trace.
+    2. **Add new key-value pair**. New key-value pair should be added into the beginning of the list.
+    3. **Delete the key-value pair**. Any key-value pair MAY be deleted. It is
+    highly discouraged to delete keys that weren't generated by the same tracing
+    system or platform. Deletion of unknown key-value pair will break correlation
+    in other systems. This mutation enables two scenarios. The first is proxies
+    can block certain `tracestate` keys for privacy and security concerns. The
+    second scenario is a truncation of long `tracestate`'s.
+
+## Processing Model for (re)starting a trace
+
+1. **Restarting trace**. All properties - `trace-id`, `parent-id`, `trace-flags`
+   are regenerated. This mutation is used in the services defined as a front
```

Comment:
"is used" => "can be used" or "is typically used". I don't think this behavior should be limited to a single scenario.

Reply:
This section has be updated in https://github.com/w3c/trace-context/pull/246/commits/25b844e5d4df816153b6147cced330cb6dd34b86 - please review

Reply:
In case you are restarting a trace you should record the old traceparent. We have not really defined this behaviour